src/authenticationservices.cs

```diff
@@ -743,4 +778,148 @@ interface ASWebAuthenticationSessionWebBrowserSessionManager {
+	interface ASAccountAuthenticationModificationRequest {
+	}
+
+	interface IASAccountAuthenticationModificationControllerDelegate {}
+
+	[iOS (14,0)]
+	[NoWatch, NoTV, NoMac]
+	[Unavailable (PlatformName.MacCatalyst)][Advice ("This API is not available when using Catalyst on macOS.")]
+	[Protocol][Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface ASAccountAuthenticationModificationControllerDelegate {
+
+		[Export ("accountAuthenticationModificationController:didSuccessfullyCompleteRequest:withUserInfo:")]
+		void DidSuccessfullyCompleteRequest (ASAccountAuthenticationModificationController controller, ASAccountAuthenticationModificationRequest request, [NullAllowed] NSDictionary userInfo);
+
+		[Export ("accountAuthenticationModificationController:didFailRequest:withError:")]
+		void DidFailRequest (ASAccountAuthenticationModificationController controller, ASAccountAuthenticationModificationRequest request, NSError error);
+	}
+
+	interface IASAccountAuthenticationModificationControllerPresentationContextProviding {}
+
+	[iOS (14,0)]
+	[NoWatch, NoTV, NoMac]
+	[Unavailable (PlatformName.MacCatalyst)][Advice ("This API is not available when using Catalyst on macOS.")]
+	[Protocol]
+	interface ASAccountAuthenticationModificationControllerPresentationContextProviding {
+
+		[Abstract]
+		[Export ("presentationAnchorForAccountAuthenticationModificationController:")]
+		UIWindow GetPresentationAnchor (ASAccountAuthenticationModificationController controller);
+	}
+
+	[iOS (14,0)]
+	[NoWatch, NoTV, NoMac]
+	[Unavailable (PlatformName.MacCatalyst)][Advice ("This API is not available when using Catalyst on macOS.")]
+	[BaseType (typeof (NSObject))]
+	interface ASAccountAuthenticationModificationController {
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		IASAccountAuthenticationModificationControllerDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+
+		[NullAllowed, Export ("presentationContextProvider", ArgumentSemantic.Weak)]
+		IASAccountAuthenticationModificationControllerPresentationContextProviding PresentationContextProvider { get; set; }
+
+		[Export ("performRequest:")]
+		void PerformRequest (ASAccountAuthenticationModificationRequest request);
+	}
+
+	[iOS (14,0)]
+	[NoWatch, NoTV, NoMac]
+	[Unavailable (PlatformName.MacCatalyst)][Advice ("This API is not available when using Catalyst on macOS.")]
+	[BaseType (typeof (NSExtensionContext))]
+	[DisableDefaultCtor]
+	interface ASAccountAuthenticationModificationExtensionContext {
+
+		[Export ("getSignInWithAppleUpgradeAuthorizationWithState:nonce:completionHandler:")]
+		void GetSignInWithAppleUpgradeAuthorization ([NullAllowed] string state, [NullAllowed] string nonce, Action<ASAuthorizationAppleIdCredential, NSError> completionHandler);
+
+		[Export ("completeUpgradeToSignInWithAppleWithUserInfo:")]
+		void CompleteUpgradeToSignInWithApple ([NullAllowed] NSDictionary userInfo);
```

Comment:
Do we need the 'WithApple'?

Reply:
Yes, we need it, because it's the name of a specific way to sign in.

Reply:
The feature name is "Sign in with Apple", so it's an all or nothing